### ISSUE TITLE
KVM without IOMMU support doesn't build

### ISSUE THREAD

**Issue by mpe** [linuxppc/issues#226]
See: http://kisskb.ellerman.id.au/kisskb/buildresult/13692649/

Happens on v5.0-rc2 ish.

```
(.text+0x1380): undefined reference to `kvmppc_h_get_tce'
(.text+0x1384): undefined reference to `kvmppc_rm_h_put_tce'
(.text+0x149c): undefined reference to `kvmppc_rm_h_stuff_tce'
(.text+0x14a0): undefined reference to `kvmppc_rm_h_put_tce_indirect'
```

Probably just requires adding some ifdefs.

**Comment by sathnaga** [comment]
@mpe I tried fixing it and compilation is working fine with this patch, let me know if fine to post in mailing list, 
[0001-powerpc-Fix-compilation-issue-without-IOMMU-Config.txt](https://github.com/linuxppc/issues/files/2936408/0001-powerpc-Fix-compilation-issue-without-IOMMU-Config.txt)

**Comment by mpe** [comment]
Thanks for the patch @sathnaga. Unfortunately this has already been fixed in Paul's kvm-ppc tree:

https://git.kernel.org/pub/scm/linux/kernel/git/paulus/powerpc.git/commit/?h=kvm-ppc-next&id=e40542aff909ac34d2c24712c5c0769c8f77f895

Sorry I didn't get around to closing the issue. On the plus side your patch looks pretty much the same :+1:

**Comment by sathnaga** [comment]
@mpe Sure, np.

**Comment by sathnaga** [comment]
@mpe learning, I should not be looking for `in-progress` one's...

**Comment by mpe** [comment]
Yeah I *try* to mark that on ones that are being worked on. I should probably just put a comment saying "Person x is working on this".

**Comment by mpe** [comment]
Fixed in torvalds/linux@e40542aff909ac34d2c24712c5c0769c8f77f895.

Thanks @iamjpn :1st_place_medal:
